src/src/main/java/service/Impl/BasketServiceImpl.java

```diff
@@ -0,0 +1,70 @@
+package service.Impl;
+
+import dao.BasketDAO;
+import factories.BasketDAOFactory;
+import model.Basket;
+import model.Product;
+import model.User;
+import service.BasketService;
+
+import java.util.List;
+import java.util.Optional;
+
+public class BasketServiceImpl implements BasketService {
+    private static final BasketDAO basketDAO = BasketDAOFactory.getInstance();
+
+    @Override
+    public List<Basket> getAllBaskets() {
+        return basketDAO.getAllBaskets();
+    }
+
+    @Override
+    public void createBasket(User user) {
+        basketDAO.createBasket(user);
+    }
+
+    @Override
+    public Optional<Basket> getUserBasket(User user) {
+        return getAllBaskets().stream()
```

Comment:
we meed to move this logic in dao layer
the same with all methods in this class

Reply:
fixed, but unfortunately, I pushed it in another branch, I changed the link in the table